src/os/bluestore/BlueStore.cc

```diff
@@ -3612,6 +3612,10 @@ void BlueStore::MempoolThread::_balance_cache(
     const std::list<PriorityCache::PriCache *>& caches)
 {
   int64_t mem_avail = autotune_cache_size;
+  /* Each cache is going to get a little extra from get_chunk, so shrink the
+   * available memory here to compensate. */
+  mem_avail -= PriorityCache::get_chunk(1, autotune_cache_size) * caches.size();
+  assert(mem_avail >= 0);
```

Comment:
```
2018-12-22T08:24:33.051 INFO:teuthology.orchestra.run.smithi132.stderr:ceph-osd: /build/ceph-14.0.1-1984-g559ba35/src/os/bluestore/BlueStore.cc:3638: void BlueStore::MempoolThread::_balance_cache(const std::__cxx11::list<PriorityCache::PriCache*>&): Assertion `mem_avail >= 0' failed.
2018-12-22T08:24:33.051 INFO:teuthology.orchestra.run.smithi132.stderr:*** Caught signal (Aborted) **
2018-12-22T08:24:33.051 INFO:teuthology.orchestra.run.smithi132.stderr: in thread 7f72f01d5700 thread_name:bstore_mempool
2018-12-22T08:24:33.053 INFO:teuthology.orchestra.run.smithi132.stderr: ceph version 14.0.1-1984-g559ba35 (559ba35765d533c579d0ba8e2bb47feedfc3f441) nautilus (dev)
2018-12-22T08:24:33.053 INFO:teuthology.orchestra.run.smithi132.stderr: 1: (()+0x12890) [0x7f72ffba3890]
2018-12-22T08:24:33.054 INFO:teuthology.orchestra.run.smithi132.stderr: 2: (gsignal()+0xc7) [0x7f72fe850e97]
2018-12-22T08:24:33.054 INFO:teuthology.orchestra.run.smithi132.stderr: 3: (abort()+0x141) [0x7f72fe852801]
2018-12-22T08:24:33.054 INFO:teuthology.orchestra.run.smithi132.stderr: 4: (()+0x3039a) [0x7f72fe84239a]
2018-12-22T08:24:33.054 INFO:teuthology.orchestra.run.smithi132.stderr: 5: (()+0x30412) [0x7f72fe842412]
2018-12-22T08:24:33.054 INFO:teuthology.orchestra.run.smithi132.stderr: 6: (BlueStore::MempoolThread::_balance_cache(std::__cxx11::list<PriorityCache::PriCache*, std::allocator<PriorityCache::PriCache*> > const&)+0x267) [0x5620728e3327]
2018-12-22T08:24:33.055 INFO:teuthology.orchestra.run.smithi132.stderr: 7: (BlueStore::MempoolThread::entry()+0x2a8) [0x5620728e3608]
2018-12-22T08:24:33.055 INFO:teuthology.orchestra.run.smithi132.stderr: 8: (()+0x76db) [0x7f72ffb986db]
2018-12-22T08:24:33.055 INFO:teuthology.orchestra.run.smithi132.stderr: 9: (clone()+0x3f) [0x7f72fe93388f]
2018-12-22T08:24:33.056 INFO:teuthology.orchestra.run.smithi132.stderr:2018-12-22 08:24:33.047 7f72f01d5700 -1 *** Caught signal (Aborted) **
2018-12-22T08:24:33.056 INFO:teuthology.orchestra.run.smithi132.stderr: in thread 7f72f01d5700 thread_name:bstore_mempool
```
http://pulpito.ceph.com/kchai-2018-12-22_08:06:28-rados-wip-kefu2-testing-2018-12-22-1018-distro-basic-smithi/

Reply:
This is likely happening in a corner case when the aggregate minimum chunk size is larger than cache_min (probably when the cache is first initialized to cache_min, but could happen in other circumstances).  Instead of asserting, we should just set mem_avail to 0.